### PR TITLE
Fix unit error in CCCVcharging

### PR DESCRIPTION
**Pull request by henrikt-ma**

### CODE REVIEW

**Review by hubertus65 — APPROVED** [review]

**Review by AHaumer — DISMISSED** [review]
I'd suggest the following change:
`  parameter Modelica.Units.SI.Current Isc = 1200 "Short-circuit current of cell at OCVmax";`
Why don't make this parameter public?

File: Modelica/Electrical/Batteries/Examples/CCCVcharging.mo [hunk 1]
```diff
@@ -2,10 +2,13 @@ within Modelica.Electrical.Batteries.Examples;
...
   "Charge a battery with constant current - constant voltage characteristic"
   extends Modelica.Icons.Example;
+protected
+  parameter Modelica.Units.SI.Current RiCurrent = 1200 "Current relating OCVmax to Ri in cellData";
+public
   parameter Modelica.Electrical.Batteries.ParameterRecords.TransientData.ExampleData cellData(
     Qnom=18000,
     useLinearSOCDependency=false,
-    Ri=cellData.OCVmax/1200,
+    Ri=cellData.OCVmax/RiCurrent,
```
Comment (HansOlsson) [inline Modelica/Electrical/Batteries/Examples/CCCVcharging.mo:11]:
```suggestion
    Ri=cellData.OCVmax/Isc,
```
Together with other change from @AHaumer

File: Modelica/Electrical/Batteries/Examples/CCCVcharging.mo [hunk 2]
```diff
@@ -2,10 +2,13 @@ within Modelica.Electrical.Batteries.Examples;
 model CCCVcharging
   "Charge a battery with constant current - constant voltage characteristic"
   extends Modelica.Icons.Example;
+protected
+  parameter Modelica.Units.SI.Current RiCurrent = 1200 "Current relating OCVmax to Ri in cellData";
```
[the remark this answers is not in the document]
Comment (henrikt-ma) [reply Modelica/Electrical/Batteries/Examples/CCCVcharging.mo:6]: Yes, why not.

**Comment by HansOlsson** [comment]
Coordinate as described in #4101

File: Modelica/Electrical/Batteries/Examples/CCCVcharging.mo [hunk 3]
```diff
@@ -2,10 +2,12 @@ within Modelica.Electrical.Batteries.Examples;
 model CCCVcharging
   "Charge a battery with constant current - constant voltage characteristic"
   extends Modelica.Icons.Example;
+  parameter Modelica.Units.SI.Current Isc = 1200 "Short-circuit current of cell at OCVmax";
+public
```
Comment (christiankral) [inline Modelica/Electrical/Batteries/Examples/CCCVcharging.mo:6]:
```suggestion
```

**Review by christiankral — COMMENTED** [review]
@henrikt-ma I suggest to remove `public`, see suggestion. 

Other than that, I am in favor of approving this proposal.

**Review by christiankral — APPROVED** [review]
`OCVmax` and `Isc` are now considered in each battery example

File: Modelica/Electrical/Batteries/Examples/CCCV_Cell.mo [hunk 4]
```diff
@@ -2,10 +2,11 @@ within Modelica.Electrical.Batteries.Examples;
 model CCCV_Cell
   "Charge a cell with constant current - constant voltage characteristic"
   extends Modelica.Icons.Example;
+    parameter Modelica.Units.SI.Current Isc = 1200 "Short-circuit current of cell at OCVmax";
```
Comment (maltelenz) [inline Modelica/Electrical/Batteries/Examples/CCCV_Cell.mo:5]: Match indentation to the remaining lines?
```suggestion
  parameter Modelica.Units.SI.Current Isc = 1200 "Short-circuit current of cell at OCVmax";
```

Comment (christiankral) [reply Modelica/Electrical/Batteries/Examples/CCCV_Cell.mo:5]: Remove blanks

File: Modelica/Electrical/Batteries/Examples/CCCV_StackRC.mo [hunk 5]
```diff
@@ -2,16 +2,17 @@ within Modelica.Electrical.Batteries.Examples;
...
     useLinearSOCDependency=false,
-    Ri=4.2/1200,
+    Ri=cellDataOriginal.OCVmax/Isc,
     Idis=0.001) "Original cell data"
     annotation (Placement(transformation(extent={{-40,-60},{-20,-40}})));
   parameter Modelica.Electrical.Batteries.ParameterRecords.TransientData.ExampleData cellDataDegraded(
     Qnom=18000,
     useLinearSOCDependency=false,
-    Ri=2*4.2/1200,
+    Ri=cellDataDegraded.OCVmax/Isc,
```
Comment (HansOlsson) [inline Modelica/Electrical/Batteries/Examples/CCCV_StackRC.mo:15]:
```suggestion
    Ri=2*cellDataDegraded.OCVmax/Isc,
```
Previously it was 2*4.2/1200, and OCVmax=4.2, Isc=1200; but the factor 2 was lost.
Without this change the I don't see any difference between original and degraded.

**Review by HansOlsson — COMMENTED** [review]
Fine by me - I think it is better to wait for actual expert approval; and I haven't figured out how to remove myself as reviewer.

File: Modelica/Electrical/Batteries/Examples/CCCVcharging.mo (hunk 3 above)
[the remark this answers is not in the document]
Comment (hubertus65) [reply Modelica/Electrical/Batteries/Examples/CCCVcharging.mo]: Resolved since outdated and agreed in meeting 2023-11-14

**Review by hubertus65 — APPROVED** [review]

**Pull request merged**